starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -85,7 +87,14 @@ public Optional<Inventory> update(String id, Inventory inventory) {
    * @return Deleted Inventory.
    */
   public Optional<Inventory> delete(String id) {
-    // TODO
+    //get Inventory by ID
+    if(retrieve(id).isPresent()){
+      Inventory ret = retrieve(id).get();
```

Comment:
Instead of retrieving the entire inventory and then deleting like that, you can delete the inventory by specifying a query and then passing the query into the remove method.

expample 
this.mongoTemplate.remove(<insert query here>, Inventory.class);


Review the following documentation on how to remove the document using a query
https://docs.spring.io/spring-data/mongodb/docs/current/api/org/springframework/data/mongodb/core/MongoTemplate.html#remove(org.springframework.data.mongodb.core.query.Query,java.lang.Class)